zookeeper-api/src/main/java/org/apache/helix/zookeeper/zkclient/IZkChildListener.java

```diff
@@ -29,17 +31,31 @@
  * guaranteed that events on the path are missing (see http://zookeeper.wiki.sourceforge.net/ZooKeeperWatches). An
  * implementation of this class should take that into account.
  *
+ * Deprecated: This interface is kept to maintain backward compatibility, please use {@link IZkChildEventListener}.
  */
-public interface IZkChildListener {
+@Deprecated
+public interface IZkChildListener extends IZkChildEventListener {
```

Comment:
This is same effect as I mentioned. If you force this extends IZkChildEventListener, the impl of new API will be missing.

Reply:
How is that? There is a default implementation of the new method. There is no breaking change for existing classes, `ZkHelixClusterVerifier` is an example